### PR TITLE
Retry 3 times when encountering Google flaky 500 error

### PR DESCRIPTION
**Pull request by theefer**
We don't know why Google sometimes returns a 500 error with a snippet of JSON helpfully describing it as "Backend Error", but I thought hey why not retry it a few times when that happens in case it's a race condition or whatnot.

Such errors are observed on all the apps using panda, according to our logging server.

Sadly don't really have a way to reproduce this, apart from running in PROD.

Happy for the Grid to be the guinea pig for this.

### CODE REVIEW

**Comment by theefer** [comment]
cc @akash1810

File: pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/service/GoogleAuth.scala [hunk 1]
```diff
@@ -99,4 +93,40 @@ class GoogleAuth(config: GoogleAuthSettings, system: String, redirectUrl: String
...
     }
   }
-}
\ No newline at end of file
+
+
+  private def requestToken(tokenEndpoint: String, code: Seq[String], retries: Int = 3): Future[WSResponse] = {
+    WS.url(tokenEndpoint).post {
+      Map(
+        "code" -> code,
```
Comment (akash1810) [inline pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/service/GoogleAuth.scala]: is there a chance `code` will expire whilst in the retry loop, which will (hopefully) throw a different error than `Backend Error`.

Comment (theefer) [inline pan-domain-auth-play/src/main/scala/com/gu/pandomainauth/service/GoogleAuth.scala]: I don't know, but if so the other error will just break the recursion and be passed to the caller?

**Comment by akash1810** [comment]
:+1:

**Pull request merged**